<h2 style="color:green" align="center"> Machine Learning With Python: Linear Regression Multiple Variables</h2>

#### Predicting the profits of 50 startups based on different variables

In [1]:
# importing the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# importing the datasets and exploring

df = pd.read_csv("50_Startups.csv")

In [3]:
# Seperating the dependent(y) and independent(x) variables
x = df.iloc[:,:-1]
y = df.iloc[:,4]

In [4]:
# converting the column into catogorical column
states = pd.get_dummies(x['State'], drop_first=True)

# dropping the state column
x = x.drop('State', axis=1)

#Concating the dummy variable
x = pd.concat([x,states], axis=1)

In [5]:
# Splitting the datasets into training and test sets
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

In [6]:
# Fitting multiple regression into training sets
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

LinearRegression()

In [7]:
# predicting the test sets result
y_predict = regressor.predict(x_test)
y_predict

array([103015.20159796, 132582.27760816, 132447.73845174,  71976.09851258,
       178537.48221055, 116161.24230165,  67851.69209676,  98791.73374687,
       113969.43533012, 167921.0656955 ])

In [8]:
from sklearn.metrics import r2_score
score = r2_score(y_test, y_predict)
score

0.9347068473282423

### From above  its clear that model is good to go as prediction is 93.5% correct.